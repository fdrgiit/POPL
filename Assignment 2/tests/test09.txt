a b c d e f #
LPAREN  RPAREN  INT  ID  SEMICOLON  NUM  REALNUM STRING COMMA DIV UNDERSCORE WHILE #
a -> LPAREN b RPAREN #
a -> d e #
b -> #
b -> b INT c SEMICOLON #
b -> b NUM c SEMICOLON #
b -> b REALNUM c SEMICOLON #
b -> b STRING c SEMICOLON #
c -> ID #
c -> c COMMA ID #
d -> #
d -> DIV DIV UNDERSCORE #
d -> WHILE d #
e -> f e #
e -> f #
f -> #     ##

