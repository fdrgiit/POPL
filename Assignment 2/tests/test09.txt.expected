FIRST(a) = { #, DIV, LPAREN, WHILE }
FIRST(b) = { #, INT, NUM, REALNUM, STRING }
FIRST(c) = { ID }
FIRST(d) = { #, DIV, WHILE }
FIRST(e) = { # }
FIRST(f) = { # }
FOLLOW(a) = { $ }
FOLLOW(b) = { INT, NUM, REALNUM, RPAREN, STRING }
FOLLOW(c) = { COMMA, SEMICOLON }
FOLLOW(d) = { $ }
FOLLOW(e) = { $ }
FOLLOW(f) = { $ }
